python/ray/data/datasource/file_based_datasource.py

```diff
@@ -644,6 +644,7 @@ def _resolve_paths_and_filesystem(
     resolved_paths = []
     for path in paths:
         path = _resolve_example_path(path)
+        path = _resolve_local_scheme(path)
```

Comment:
Can we unify the custom scheme handling for the example + local schemes? It seems like they should have the same mechanism.

Reply:
Done.